### ISSUE TITLE
Move website source to main repository and add markdown template expansion support

### ISSUE THREAD

**Issue by jsichi** [jgrapht/jgrapht#584]
Here is my plan per discussions in https://groups.google.com/forum/#!topic/jgrapht-dev/IsD6DInfNAE

I may do some of these steps directly rather than via pull request since some things are hard to test without a deployed site.

- [x] Copy sources from jgrapht/jgrapht.github.com repository (including javadoc for latest release) to new docs directory and add Jekyll Gemfile
- [x] Update .travis.yml to deploy site to gh-pages branch (during testing, this will be served only from fruitbat.jgrapht.org)
- [x] Update .travis.yml to expand templates as part of deployment
- [x] Update .travis.yml to produce SNAPSHOT Javadoc as part of deployment (release Javadoc will remain untouched)
- [x] Futz around with https since I think we'll be forced to enable it, and this may require us to change the way some URL's are generated in Javadoc etc
- [x] Write a wiki page documenting the website deployment, including how to test changes locally
- [x] Stop serving www.jgrapht.org from jgrapht/jgrapht.github.com repository and start serving it from jgrapht/jgrapht repository instead
- [x] Archive old jgrapht/jgrapht.github.com repository

Note that this issue does not include reworking the website content or writing the new user guide; those will be undertaken after this issue is resolved.

**Comment by jsichi** [comment]
Doc here:

https://github.com/jgrapht/jgrapht/wiki/Website-Deployment

**Comment by jkinable** [comment]
Looks good!

- Could you update `https://github.com/jgrapht/jgrapht/blob/master/etc/release-process.html` where necessary?
- In the wiki description, I couldn't find how to add tags to the demo classes for code snippets. I couldn't find the script you used for this either. Perhaps I overlooked it.

**Comment by jsichi** [comment]
I've updated the release process doc and converted it to markdown:

https://github.com/jgrapht/jgrapht/blob/master/etc/ReleaseProcess.md

And I've added a page on how to add the directive comments here:

https://github.com/jgrapht/jgrapht/wiki/How-to-add-example-code

The expansion script is now linked from the bottom of that page.

I'm planning to go live on www with the new infrastructure tonight.

**Comment by jsichi** [comment]
Transition complete.  Rather than deleting the old jgrapht.github.com repository, I put it in archive mode; we can preserve it since it has some edit history for the website.

For enabling HTTPS, I had to update the javadoc build to use https for the mathjax script link (to avoid mixed content errors).  I've manually applied this change to all versions in the jgrapht-javadoc repository as well, so now mathjax gets rendered correctly on the site.
